debugger/src/debug_adapter/repl_commands.rs

```diff
@@ -0,0 +1,774 @@
+    DapReference,
+    /// Native (as defined by data type) format.
+    Native,
+}
+
+impl TryFrom<&char> for GdbFormat {
+    type Error = DebuggerError;
+
+    fn try_from(format: &char) -> Result<Self, Self::Error> {
+        match format {
+            't' => Ok(GdbFormat::Binary),
+            'x' => Ok(GdbFormat::Hex),
+            'i' => Ok(GdbFormat::Instruction),
+            'v' => Ok(GdbFormat::DapReference),
+            'n' => Ok(GdbFormat::Native),
+            _ => Err(DebuggerError::ReplError(format!(
+                "Invalid format specifier: {format}"
+            ))),
+        }
+    }
+}
+
+impl Display for GdbFormat {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            GdbFormat::Binary => write!(f, "t(binary)"),
+            GdbFormat::Hex => write!(f, "x(hexadecimal)"),
+            GdbFormat::Instruction => write!(f, "i(nstruction)"),
+            GdbFormat::DapReference => write!(f, "v(ariable structured for DAP/VSCode)"),
+            GdbFormat::Native => write!(f, "n(ative - as defined by data type)"),
+        }
+    }
+}
+
+enum GdbUnit {
+    Byte,
+    HalfWord,
+    Word,
+    Giant,
+}
+
+impl TryFrom<&char> for GdbUnit {
+    type Error = DebuggerError;
+
+    fn try_from(unit_size: &char) -> Result<Self, Self::Error> {
+        match unit_size {
+            'b' => Ok(GdbUnit::Byte),
+            'h' => Ok(GdbUnit::HalfWord),
+            'w' => Ok(GdbUnit::Word),
+            'g' => Ok(GdbUnit::Giant),
+            _ => Err(DebuggerError::ReplError(format!(
+                "Invalid unit size: {unit_size}"
+            ))),
+        }
+    }
+}
+
+impl GdbUnit {
+    fn get_size(&self) -> usize {
+        match self {
+            GdbUnit::Byte => 1,
+            GdbUnit::HalfWord => 2,
+            GdbUnit::Word => 4,
+            GdbUnit::Giant => 8,
+        }
+    }
+}
+
+impl Display for GdbUnit {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            GdbUnit::Byte => write!(f, "b(yte)"),
+            GdbUnit::HalfWord => write!(f, "h(alfword)"),
+            GdbUnit::Word => write!(f, "w(ord)"),
+            GdbUnit::Giant => write!(f, "g(iant)"),
+        }
+    }
+}
+
+struct GdbNuf {
+    unit_count: usize,
+    unit_specifier: GdbUnit,
+    format_specifier: GdbFormat,
+}
+
+impl GdbNuf {
+    // TODO: If the format_specifier is `instruction` we should return the size of the instruction for the architecture.
+    fn get_size(&self) -> usize {
+        self.unit_count * self.unit_specifier.get_size()
+    }
+    // Validate that the format specifier is valid for a given range of supported formats
+    fn check_supported_formats(&self, supported_list: &[GdbFormat]) -> Result<(), String> {
+        if supported_list.contains(&self.format_specifier) {
+            Ok(())
+        } else {
+            let mut message = "".to_string();
```

Comment:
This returns an empty string if this is called with an empty list, it would be better to get a helpful error message in that case.

Reply:
Done :)